### ISSUE TITLE
[Bug]: DetailsList How to make certain columns fixed on the right

### ISSUE THREAD

**Issue by ChangBaiMonster** [microsoft/fluentui#24785]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
React v7
fluent ui v8
```

### Are you reporting Accessibility issue?

yes

### Reproduction

no

### Bug Description

The current phenomenon is that my DetailsList has many operation columns, so the width adaptive scheme cannot be used. When there are many columns, the horizontal scroll bar will inevitably appear. So I added a filter column function to the DetailsList. When there are few columns left, I hope the last operation column is fixed on the right, and the other columns can adapt to the width.
Each item in columns will be set to minWidth, maxWidth.  I use the columnReorderOptions = { {frozenColumnCountFromEnd:0 }} ,but doesn't work.

## Actual Behavior
But now, when there are only two columns left, they are stacked together, and the last column cannot be fixed on the right

## Expected Behavior
 when there are only two columns left,  the last column can be fixed on the right. The front column can adapt to the remaining width

### Logs

```shell
I want some of the columns to be fixed on the right, but this column can still be resize, and then when the other left columns are too long, a scroll bar on the left appears。


<DetailsList
items={items}
compact={isCompactMode}
columns={columns}
selectionMode={SelectionMode.none}
getKey={this._getKey}
setKey="none"
layoutMode={DetailsListLayoutMode.fixed}
isHeaderVisible={true}
onItemInvoked={this._onItemInvoked}
columnReorderOptions = { {frozenColumnCountFromEnd:0 }}
/>
```

### Requested priority

High

### Products/sites affected

no

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
Thanks for filing this issue. We are delighted to see that you are willing to contribute to the reported issue, but you’ll need to add a CodePen repro before we can help. You can start from one of the [DetailsList ](https://developer.microsoft.com/en-us/fluentui?fabricVer=8#/controls/web/detailslist) examples (click “Export to CodePen”) or https://aka.ms/fluentpen.

**Comment by ChangBaiMonster** [comment]
> Thanks for filing this issue. We are delighted to see that you are willing to contribute to the reported issue, but you’ll need to add a CodePen repro before we can help. You can start from one of the [DetailsList ](https://developer.microsoft.com/en-us/fluentui?fabricVer=8#/controls/web/detailslist) examples (click “Export to CodePen”) or https://aka.ms/fluentpen.

I simulated an example of switching columns, and I wanted the action column to be fixed to the right and the other columns to appear scrollbars
here is an example:
https://codepen.io/wqwdna/pen/ZEorGNd

**Comment by micahgodbolt** [comment]
Questions are better asked in our discussion section. Issues are for items that need to be resolved.
